**ДЗ семинар 6.   
Сравнение долей. Построение доверительных интервалов.**

*Задача 1:    
Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.*

Доверительный интервал найдем по формуле $\bar X$ ± $\displaystyle Z_{α/2}$*$\frac{σ}{\sqrt[ ]{n}}$,    
где $\bar X$ - выборочная средняя M = 80, n = 256, σ = 16 и α = (1 - 0,95) = 0,05   
$\displaystyle Z_{α/2}$ находим по Z-таблице равным 1,96 
   
Доверительный интервал для 95% вероятности $\left[80-1.96\cdot\frac{16}{\sqrt 256};80+1.96\cdot\frac{16}{\sqrt 256}\right]$ 

In [8]:
left = 80 - 1.96*16/256 ** 0.5
right = 80 + 1.96*16/256 ** 0.5
print(f"Доверительный интервал для оценки математического ожидания с надежностью 0.95: [{left}; {right}]")

Доверительный интервал для оценки математического ожидания с надежностью 0.95: [78.04; 81.96]



*Ответ: [78.04; 81.96]*  

    
    

*Задача 2: 
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:   
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1    
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.*

Т.к.σ неизвестна, доверительный интервал найдем по формуле: $\bar X$ ± $\displaystyle t_{α/2}$*$\frac{σ_{несмещ.}}{\sqrt[ ]{n}}$,    
где $\bar X$ - выборочная средняя из имеющихся опытных данных, n = 10 и α = (1 - 0,95) = 0,05   

Найдем $\bar X$ выборки 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1   

In [17]:
import numpy as np
import scipy.stats as stats

In [20]:
a = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = 10 
alfa = 0.05

In [21]:
x1 = np.mean(a)
x1

6.590000000000001

Найдем несмещённое среднеквадратичное отклонение $\displaystyle σ_{несмещ.}$ выборки 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 

In [22]:
sigma = np.std(a, ddof=1)
sigma

0.4508017549014448

Значение t найдём по таблице для n = 9 и α = 0,05    
t1 = 2.262    
Или найдем с помощью функции: 

In [23]:
t1 = stats.t.ppf (1-alfa/2, n-1)
t1

2.2621571627409915

Подставим все известные значения в формулу: 

In [31]:
left = x1 - t1*sigma/n ** 0.5
right = x1 + t1*sigma/n ** 0.5
print(f'Доверительный интервал для вероятности 0,95: [{left:.4f}; {right:.4f}]')

Доверительный интервал для вероятности 0,95: [6.2675; 6.9125]


*Ответ: [6.2675; 6.9125]*

*Задача 3:      
Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175    
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170   
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.*

Доверительный интервал для разности средних значений найдем по формуле: $Δ$ ± $\displaystyle t_{α/2}$*$S_Δ$, где    
$Δ$ - разность между средними арифметическимы выборок роста матерей и роста дочерей  
Стандартная ошибка разности средних вычисляется по формуле $S_Δ$ = $\displaystyle \sqrt {\frac {D_1}{n_1} + \frac {D_2}{n_2}}$, где $n_1$ $n_2$ - объёмы выборок, а $D_1$ и $D_2$ - дисперсии выборок 


In [56]:
m = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])   # выборка по матерям
d = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])   # выборка по дочерям

Вычислим средние арифметические выборок и $Δ$: 

In [57]:
x1 = np.mean(m)
x2 = np.mean(d)
delta = x1-x2
print(f'Среднее арифметическое по выборке матерей {x1}\n' f'Среднее арифметическое по выборке дочерей {x2}\n' f'Δ = {delta:.1f}')

Среднее арифметическое по выборке матерей 168.1
Среднее арифметическое по выборке дочерей 166.2
Δ = 1.9


Вычислим $D_1$, $D_2$ и $S_Δ$:

In [58]:
D1 = np.var(m, ddof=1)
D2 = np.var(d, ddof=1)
SE = (D1/len(m) + D2/len(d)) ** 0.5
print(f'Стандартная ошибка разности средних SΔ = {SE:.4f}')

Стандартная ошибка разности средних SΔ = 3.8880


Найдём значение $\displaystyle t_{α/2}$ по таблице или вычислим по формуле: 

In [59]:
alfa = 0.05
n = 2*(len(m) - 1)
t = stats.t.ppf(1-alfa/2, n)
print(f'Значение t = {t:.4f}')

Значение t = 2.1009


Найдём интервал используя полученные значения: 

In [60]:
left = delta - t*SE
right = delta + t*SE
print(f"Доверительный интервал разности среднего роста матерей и дочерей для вероятности 0,95: [{left:.4f}; {right:.4f}]")

Доверительный интервал разности среднего роста матерей и дочерей для вероятности 0,95: [-6.2684; 10.0684]


*Ответ: [-6.2684; 10.0684]*